plugin/src/main/java/io/jenkins/plugins/casc/model/CNode.java

```diff
@@ -28,6 +28,8 @@ default Scalar asScalar() throws ConfiguratorException {
         throw new ConfiguratorException("Item isn't a Scalar");
     }
 
+    /** @deprecated sensitive data are identified based on target attribute being a ${@link hudson.util.Secret} */
```

Comment:
TBF, it makes sense to consider non-`Secret` types secret. I don't know what the purpose of this API is, but allowing to override it for non-`Secret` secrets seems useful.

Reply:
CNode do only reflect the yaml hierarchical data structure, so isn't relevant to distinguish some jenkins components to use an alternate way to manage sensitive data.
We used this in pre-1.0 to detect `${XX}` used as values, but the actual use of this is wider than just sensitive data replacement : many users rely on this to define a general purpose yaml configuration and customize actual values at runtime.